Source/WebCore/platform/graphics/skia/ImageBufferSkiaAcceleratedBackend.cpp

```diff
@@ -108,6 +108,33 @@ ImageBufferSkiaAcceleratedBackend::~ImageBufferSkiaAcceleratedBackend()
 #endif
 }
 
+void ImageBufferSkiaAcceleratedBackend::finishAcceleratedRenderingAndCreateFence()
+{
+    auto* glContext = PlatformDisplay::sharedDisplay().skiaGLContext();
+    if (!glContext || !glContext->makeContextCurrent())
+        return;
+
+    auto* grContext = PlatformDisplay::sharedDisplay().skiaGrContext();
+    RELEASE_ASSERT(grContext);
+
+    if (GLFence::isSupported()) {
+        grContext->flushAndSubmit(m_surface.get(), GrSyncCpu::kNo);
+        m_fence = GLFence::create();
+        if (!m_fence)
+            grContext->submit(GrSyncCpu::kYes);
+    } else
+        grContext->flushAndSubmit(m_surface.get(), GrSyncCpu::kYes);
+}
+
+void ImageBufferSkiaAcceleratedBackend::waitForAcceleratedRenderingFenceCompletion()
```

Comment:
Can't be just `waitForFenceCompletion()`?

`Accelerated` is already in the class name.

Reply:
Indeed, it's redundant, will adapt.

Reply:
Thought about this a second time: it's also part of the public ImageBuffer API, and there is no `Accelerated` in that class name. Currently the naming is symmetric:

`finishAcceleratedRenderingAndCreateFence()`
`waitForAcceleratedRenderingFenceCompletion()`

indicating that these two methods are used in combination.
I could rename one to `waitForFenceCompletion`, but not sure how to properly name the other - `finishRenderingAndCreateFence`?  I'm staying with the current names, unless you have a better suggestion :-)